scripts/linux/consolidate-build-artifacts.sh

```diff
@@ -107,18 +111,36 @@ edge-hub)
     cp -r "$EDGEHUB_ARTIFACTS_SOURCE" "$ARTIFACTS_DEST"
 
     # copy mqtt artifacts
-    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_AMD64_MUSL/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_AMD64_MUSL/release"
-    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM32V7/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_ARM32V7/release"
-    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM64V8/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_ARM64V8/release"
-    cp "$MQTT_ARTIFACTS_SOURCE/contrib/edgehub/broker.json" "$ARTIFACTS_DEST/mqtt"
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_AMD64_MUSL/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_AMD64_MUSL/release" || true
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM32V7/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_ARM32V7/release" || true
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM64V8/release/mqttd" "$ARTIFACTS_DEST/mqtt/$TARGET_ARM64V8/release" || true
+    cp "$MQTT_ARTIFACTS_SOURCE/contrib/edgehub/broker.json" "$ARTIFACTS_DEST/mqtt" || true
 
     # copy watchdog artifacts
-    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_AMD64_MUSL/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_AMD64_MUSL/release"
-    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_ARM32V7/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_ARM32V7/release"
-    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_ARM64V8/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_ARM64V8/release"
+    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_AMD64_MUSL/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_AMD64_MUSL/release" || true
+    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_ARM32V7/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_ARM32V7/release" || true
+    cp "$WATCHDOG_ARTIFACTS_SOURCE/target/$TARGET_ARM64V8/release/watchdog" "$ARTIFACTS_DEST/watchdog/$TARGET_ARM64V8/release" || true
     cp -r "$EDGEHUB_DOCKER_SOURCE" "$ARTIFACTS_DEST"
     ;;
 
+mqttd)
+    MQTT_ARTIFACTS_SOURCE="mqtt"
+    ARTIFACTS_DEST="mqtt/build-context"
+
+    # make build context structure
+    mkdir "$ARTIFACTS_DEST"
+    mkdir -p "$ARTIFACTS_DEST/$TARGET_AMD64_MUSL/release"
+    mkdir -p "$ARTIFACTS_DEST/$TARGET_ARM32V7/release"
+    mkdir -p "$ARTIFACTS_DEST/$TARGET_ARM64V8/release"
+
+    # copy mqtt artifacts
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_AMD64_MUSL/release/mqttd" "$ARTIFACTS_DEST/$TARGET_AMD64_MUSL/release" || true
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM32V7/release/mqttd" "$ARTIFACTS_DEST/$TARGET_ARM32V7/release" || true
+    cp "$MQTT_ARTIFACTS_SOURCE/target/$TARGET_ARM64V8/release/mqttd" "$ARTIFACTS_DEST/$TARGET_ARM64V8/release" || true
```

Comment:
This ```|| true``` pattern lets the step fail. That is useful for local builds, because you don't want to build the rust artifacts for every arch. It is also useful for this mqtt image build because they have separate jobs that build for each arch

I had a misunderstanding from before where I thought it wouldn't fail out because I removed the ```set error -e``` from the top of the script. Turns out it still does and we need || true.  This won't cause any issues because if artifacts are truly missing the docker build will fail